packages/block-editor/src/hooks/child-layout.js

```diff
@@ -0,0 +1,172 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalToggleGroupControl as ToggleGroupControl,
+	__experimentalToggleGroupControlOption as ToggleGroupControlOption,
+	__experimentalUnitControl as UnitControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import useSetting from '../components/use-setting';
+
+function helpText( selfStretch ) {
+	switch ( selfStretch ) {
+		case 'fill':
+			return __( 'Stretch to fill available space.' );
+		case 'fixed':
+			return __( 'Specify a fixed width.' );
+		default:
+			return __( 'Fit contents.' );
+	}
+}
+
+/**
+ * Inspector controls containing the child layout related configuration.
+ *
+ * @param {Object} props                        Block props.
+ * @param {Object} props.attributes             Block attributes.
+ * @param {Object} props.setAttributes          Function to set block attributes.
+ * @param {Object} props.__unstableParentLayout
+ *
+ * @return {WPElement} child layout edit element.
+ */
+export function ChildLayoutEdit( {
```

Comment:
Very minor naming nit (feel free to ignore this one!): now that these controls are injected into Dimensions instead of the layout panel, would it be worth clarifying the intended use by calling this function `ChildLayoutDimensionsEdit` or something like that?

Reply:
I'm not too fussed either way, but given there's been a bit of discussion about unifying Dimensions and Layout, maybe we can leave it as is until we reach a decision there 😅

Reply:
> I'm not too fussed either way, but given there's been a bit of discussion about unifying Dimensions and Layout, maybe we can leave it as is until we reach a decision there 😅

Sounds good to me! It's not a component we're exporting outside of this package, so it'll be easy to change in follow-ups 🙂